test/gcext/gcext.c

```diff
@@ -307,6 +307,7 @@ static size_t gc_alloc_size(jl_value_t *val)
 
 int internal_obj_scan(jl_value_t *val)
 {
+    // FIXME: `jl_gc_internal_obj_base_ptr` is not allowed to be called from outside GC
```

Comment:
@rbehrends we already register a task scanner and a root scanner callback in gcext, which exercise `jl_gc_internal_obj_base_ptr`. So could we just remove `internal_obj_scan`? (I guess if we did that, a major part of `gcext.c` and `LocalTest.jl` would end up dead code and could be removed...)

Or do you think it's still valuable? If so, perhaps the code @vchuravy added here in a previous revision of the PR (and which is still visible in `https://github.com/JuliaLang/julia/pull/47407/files`)  which sets/restores `gc_n_threads` and `gc_all_tls_states`, could simply be inserted into `internal_obj_scan` ?

Reply:
The problem is more that you are looking at data that is managed by the GC without holding the GC "lock".  So the data might be mutated underneath you. So if we could move this to inside the `mark_stack` that would be  better.